recommender/migrate/utils.py

```diff
@@ -0,0 +1,168 @@
+    return migration_class
+
+
+def scan_for_migrations():
+    """
+    Scans for migrations in the MIGRATE_DIR and
+    then inserts newly found migrations into the Migration collection.
+    """
+
+    print("Scanning for new migrations...")
+    names_from_migrations_dir = _get_names_from_migrations_dir()
+    names_from_db = [migration.name for migration in Migration.objects()]
+    missing_names = set(names_from_migrations_dir) - set(names_from_db)
+
+    if missing_names:
+        for missing_name in missing_names:
+            print(
+                f"\tInserting new migration {missing_name} into the Migration collection..."
+            )
+            new_migration_model = Migration(name=missing_name, applied=False)
+            new_migration_model.save()
+
+        print("\tNew migrations acknowledged!")
+        return True
+
+    print("\tFound no new migrations!")
+    return False
+
+
+def apply_migrations():
+    """Applies unapplied migrations"""
+
+    print("Applying migrations...")
+
+    unapplied_migrations = Migration.objects(applied=False).order_by("+name")
+
+    if unapplied_migrations:
+        for unapplied in unapplied_migrations:
+            print(f"\tApplying `{unapplied.name}` ...")
+
+            # Retrieve class
+            migration_class = _get_migration_class(unapplied)
+
+            # Apply migration
+            migration_class().up()
+
+            # Update migration document
+            unapplied.update(applied=True)
+
+        print("\tDone applying migrations!")
+        return True
+
+    print("\tNo unapplied migrations found!")
+    return False
+
+
+def rollback_last_migration():
+    """Rollbacks last applied migration"""
+
+    print("Rollbacking last applied migration...")
+
+    applied_migrations = Migration.objects(applied=True).order_by("-name")
+
+    if applied_migrations:
+        last_applied = applied_migrations.first()
+        print(f"\tRollbacking `{last_applied.name}` ...")
+
+        # Retrieve class
+        migration_class = _get_migration_class(last_applied)
+
+        # Rollback migration
+        migration_class().down()
+
+        # Update migration document
+        last_applied.update(applied=False)
+
+        print("\tDone rollbacking!")
+        return True
+
+    print("\tThere are no applied migrations to rollback!")
+    return False
+
+
+def list_migrations():
+    """Lists migrations along with their application status"""
+
+    print("Listing migrations...")
+
+    migrations = Migration.objects().order_by("+name")
+
+    if migrations:
+        for migration in migrations:
+            print(
+                f"\t {migration.name} ... {'applied' if migration.applied else 'not applied'}"
+            )
+        return True
+
+    print("\tNo migrations found!")
+    return False
+
+
+def check_migration_integrity():
+    """
+    Checks if migrations in the Migration collection
+    are the same as in the MIGRATE_DIR
+    """
```

Comment:
This function doesn't do exactly what states in its docstring. Actually it checks if migrations from cache have file representations in the migration dir. But this function doesn't check if all files from migration dir have representation in the cache. Of course new files with new migrations shouldn't have cache representations yet, but It is possible to mistakenly add a migration file with the wrong date and place it not at the end of the chronologically ordered migrations, but somewhere in the middle. Then this migration integrity check will pass when it should not. This migration that has been put in the middle is not integral with cached migrations.

Reply:
I also added to the README.md section about:

Command sequence recommendation:
- `flask migrate list`
- `flask migrate check`
- `flask migrate apply`

Reply:
Well, if the code reflects the desired use case and now the docstring has been changed accordingly, there is no need for changes.